src/Dialogs/Imap/ImapDialog.vala

```diff
@@ -374,6 +372,16 @@ public class OnlineAccounts.ImapDialog : Gtk.Window {
         });
     }
 
+    // Workaround vala memory leak
+    private void go_back () {
+        navigation_view.pop ();
+    }
+
+    // Workaround vala memory leak
+    private void push_page (Adw.NavigationPage page) {
```

Comment:
That's not really necessary the only important thing is that navigation_view isn't a local variable. So by not using lambdas you kinda force yourself to not use local variables but lambdas aren't the actual source of the problem.